core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -1060,18 +1060,18 @@ object IO extends Serializable {
    * Imports an asynchronous effect into a pure `IO` value. This formulation is
    * necessary when the effect is itself expressed in terms of `IO`.
    */
-  final def asyncPure[E, A](register: (Callback[E, A]) => IO[Nothing, Unit]): IO[E, A] =
+  final def asyncPure[E, A](register: (IO[E, A] => Unit) => IO[Nothing, Unit]): IO[E, A] =
```

Comment:
This looks like one of the places where an existential could be used in place of `Unit`. Though I feel like it was mentioned before in the original issue, I feel like I might end up forgetting to flatten a `IO[IO[_]]` 😃

Reply:
Indeed this is the main drawback of the existential change. I got caught in this trap myself! I wonder if we could use negative types to error out if the `IO` contains another `IO`?